### ISSUE TITLE
NullPointerException when clicking beatmap node, caused by scores in db

### ISSUE THREAD

**Issue by yugecin** [itdelatrisu/opsu#219]
Stacktrace points to this line:
https://github.com/itdelatrisu/opsu/blob/3b847d088d2a7d385a2b76efbaa0d81e265571cb/src/itdelatrisu/opsu/db/ScoreDB.java#L372
Which can indeed cause a npe if something weird is happening with the scores.
I've seen this at least twice in the past week, when a currupted score gets added, the game will always crash when selecting that map.

**Comment by itdelatrisu** [comment]
Should be fixed in 5920466, since the only way this could crash at that line is if the version is blank.

Do you have an example of a "corrupted score"?  (And maybe compare the score metadata with the beatmap metadata?)  That'd be helpful to have.

**Comment by yugecin** [comment]
Found it,
simple cause: if the map difficulty name is empty string, then it didn't create a new list because the if condition checked against an empty string.
So whenever you select a map with no difficulty name, it would throw a null pointer. I made some maps to test the slider things I did, and didn't bother giving it names so I had such maps in the list.
(I didn't realise version meant difficulty.)
